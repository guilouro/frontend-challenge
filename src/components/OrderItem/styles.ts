import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  width: 100%;
`

export const Image = styled.div`
  ${({ theme }) => css`
    margin-right: ${theme.spacings.xsmall};
  `}
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

export const Title = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`

export const Info = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;

    span {
      color: ${theme.colors.textLight};
    }
  `}
`
