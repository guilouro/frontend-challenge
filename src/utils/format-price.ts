export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('de-DE', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
