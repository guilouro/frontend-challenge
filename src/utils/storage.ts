import { CartObject } from 'hooks/use-cart'

const APP_KEY = 'BARKYN_CART'

export function getStorageItem() {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(APP_KEY)
  return JSON.parse(data || '[]')
}

export function setStorageItem(value: CartObject) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(APP_KEY, data)
}
