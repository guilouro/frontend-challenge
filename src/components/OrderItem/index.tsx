import Image from 'next/image'
import formatPrice from 'utils/format-price'
import * as S from './styles'

export type OrderItemProps = {
  title: string
  image: string
  price: number
}

const OrderItem = ({ title, image, price }: OrderItemProps) => (
  <S.Wrapper>
    <S.Image>
      <Image src={image} width={64} height={64} alt={title} loading="lazy" />
    </S.Image>

    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Info>
        <span>Price</span>
        <strong>{formatPrice(price)}</strong>
      </S.Info>
    </S.Content>
  </S.Wrapper>
)

export default OrderItem
