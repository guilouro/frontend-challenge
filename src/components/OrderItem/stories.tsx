import { Story, Meta } from '@storybook/react'
import OrderItem, { OrderItemProps } from '.'

export default {
  title: 'OrderItem',
  component: OrderItem,
  args: {
    title: '2x Barkyn Cannabis Oil para c√£es',
    image: '/img/products/oil.png',
    price: 4.58
  }
} as Meta

export const Default: Story<OrderItemProps> = (args) => (
  <div style={{ maxWidth: 400 }}>
    <OrderItem {...args} />
  </div>
)
