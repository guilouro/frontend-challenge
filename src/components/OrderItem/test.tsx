import 'intl.mock'
import { render, screen } from 'utils/test-utils'

import OrderItem from '.'

const props = {
  title: '2x Barkyn Cannabis Oil para c√£es',
  image: '/img/products/oil.png',
  price: 4.58
}

describe('<OrderItem />', () => {
  it('should render correctly', () => {
    const { container } = render(<OrderItem {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: props.title })).toBeInTheDocument()

    expect(screen.getByText(props.price)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
