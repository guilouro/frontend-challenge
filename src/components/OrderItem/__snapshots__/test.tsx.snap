// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItem /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  margin-right: 1.6rem;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-size: 1.6rem;
  font-weight: 400;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 span {
  color: #6c6b76;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="2x Barkyn Cannabis Oil para cães"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </div>
  </div>
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
      2x Barkyn Cannabis Oil para cães
    </h3>
    <div
      class="c4"
    >
      <span>
        Price
      </span>
      <strong>
        4.58
      </strong>
    </div>
  </div>
</div>
`
